examples/auth/gaia_auth_example.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Example for authenticating with Gaia TAP service
+"""
+import getpass
+
+import requests
+
+import pyvo
+import pyvo.extensions.auth.authsession
+
+# Gather login information
+data = {
+    'username': input('Username:'),
+    'password': getpass.getpass('Password:')
+}
+
+headers = {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Accept': 'text/plain'
+}
+
+# Create a session and do the login.
+# The cookie will end up in the cookie jar of the session.
+login_url = 'http://gea.esac.esa.int/tap-server/login'
+session = requests.Session()
+response = session.post(login_url, data=data, headers=headers)
+response.raise_for_status()
+
+# Use this session with the auth cookie for all requests to Gaia.
```

Comment:
It's relatively easy to use the cookies jar in this case:
```
service = pyvo.dal.TAPService('http://gea.esac.esa.int/tap-server/tap', auth={'SSO_COOKIE': session.cookies})
```
`session.cookies` is a `CookieJar` which should contain the newly acquired GAIA cookie. Disclaimer: I haven't tested it.

There might be other methods to pass in your own `CookieJar` in `requests.get` and have it augmented with new cookies without creating the session first but I couldn't make it work.

Reply:
Right, that's what I'm doing here right now.  I'm passing in the session, and its cookie jar, which has the cookie.

I also have the set_cookie function in credstore.  I could make a function that takes a cookie jar, that is also good.

Reply:
`set_cookie` (and optionally adding `set_cookie_jar`) in `CredStore` that could handle cookie values and cookie jars is more consistent and therefore preferable.

Reply:
Okay I'll add a set_cookie_jar as a function to the credstore